### DISCUSSION TITLE
Why is the interface of the local library gone

### DISCUSSION THREAD

**Discussion by 1641585051** [bnpr/Malt#454] [Q&A]
The latest version of the native library interface is gone, and I can't load custom glsl files and functions

**Comment by pragma37** [comment]
*Local Libraries* were replaced by plugins.

You can setup *Global Plugins* on the addon settings:
![imagen](https://user-images.githubusercontent.com/36610999/206863386-5c8d0c56-169e-4b9a-9d45-f3875738f311.png)

Or world-specific plugins in the *Pipeline Settings*:
![imagen](https://user-images.githubusercontent.com/36610999/206863456-1fbf880f-7d49-43e0-80fc-04b0c69f3c53.png)

There's a basic example plugin in the *Plugins* folder: https://github.com/bnpr/Malt/tree/Development/plugins/PluginExample
You can just drop GLSL files inside the *Shaders* folder and they should be available as nodes.

**IMPORTANT!** The *Global/Local Plugins* path shouldn't point to the *PluginExample* folder itself, but it's parent folder.
So, for example, if the your plugin path is something like `C:\MaltPlugins\MyPlugin`, the path should point to `C:\MaltPlugins`.
That also means that you can have multiple plugins in `C:\MaltPlugins`.

**Reply by 1641585051** [reply]
Ok, thanks, but I still find the local library more convenient😭
